### DISCUSSION TITLE
WASM project AAD B2C fails auth but works on Windows head

### DISCUSSION THREAD

**Discussion by skanvk15** [unoplatform/uno#9406] [Q&A]
### Current behavior

On the call to AcquireTokenInteractive the separate auth window opens to collect username and password. 

<img src="https://user-images.githubusercontent.com/51756883/179382368-2ae445e0-ecdc-4438-9f85-e0e4ae0cf9a9.png" alt="drawing" width="300"/>

After entering valid username and password the next window gets stuck with progress running forever

<img src="https://user-images.githubusercontent.com/51756883/179382382-2f40ad63-114f-4a0a-8c97-bd6b1a2f124a.png" alt="drawing" width="300"/>

Here is the logic invoked from the app:
```
                    var result = await this.publicClientApp
                        .AcquireTokenInteractive(Authentication.Scopes)
                        .WithPrompt(Prompt.ForceLogin)
                        .WithUnoHelpers()
                        .ExecuteAsync();
```

### Expected behavior

The call to AcquireTokenInteractive should complete successfully and return so app can transition to next window.

This fails using WASM head but works using Windows head!

### How to reproduce it (as minimally and precisely as possible)

To repro you will need to create a Azure AD B2C resources and project following this:
https://ian.bebbs.co.uk/posts/UnoB2C
On the Azure AD B2C policy creation step, note that I am using Display Name and not Given Name (but that should not have any impact here).

I have attached the .ZIP with the project (note only WASM and Window platforms are updated, you can ignore other platforms as that is not relevant here)

[AuthAADB2CTest.zip](https://github.com/unoplatform/uno/files/9127006/AuthAADB2CTest.zip)

1. You will need to fill in values for Authentication.Secrets.cs for using AAD B2C
2. Set the WASM project as the start project
3. Build and run
4. Sign up/in

Expectation is after sign in you will see the screen with the users display name obtained from AAD.

### Workaround

Use the Windows platform head and it works!

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

```
Id                                          Versions         ProjectName           
--                                          --------         -----------           
Microsoft.Windows.Compatibility             {5.0.0}          AuthAADB2CTest.Wasm   
System.IdentityModel.Tokens.Jwt             {6.21.0}         AuthAADB2CTest.Wasm   
Uno.WinUI.WebAssembly                       {4.4.13}         AuthAADB2CTest.Wasm   
Uno.Wasm.Bootstrap.DevServer                {3.3.1}          AuthAADB2CTest.Wasm   
Uno.WinUI.MSAL                              {4.4.13}         AuthAADB2CTest.Wasm   
Microsoft.Identity.Client                   {4.45.0}         AuthAADB2CTest.Wasm   
Uno.Wasm.Bootstrap                          {3.3.1}          AuthAADB2CTest.Wasm   
Uno.UI.Adapter.Microsoft.Extensions.Logging {4.4.13}         AuthAADB2CTest.Wasm   
Uno.WinUI.RemoteControl                     {4.4.13}         AuthAADB2CTest.Wasm   
Uno.Extensions.Logging.WebAssembly.Console  {1.3.0}          AuthAADB2CTest.Wasm   
Microsoft.Extensions.Logging                {5.0.0}          AuthAADB2CTest.Wasm   
Microsoft.Windows.SDK.BuildTools            {10.0.22000.196} AuthAADB2CTest.Windows
System.IdentityModel.Tokens.Jwt             {6.21.0}         AuthAADB2CTest.Windows
Uno.Core.Extensions.Logging.Singleton       {4.0.1}          AuthAADB2CTest.Windows
Uno.WinUI.MSAL                              {4.4.13}         AuthAADB2CTest.Windows
Microsoft.Identity.Client                   {4.45.0}         AuthAADB2CTest.Windows
Microsoft.Extensions.Logging.Console        {5.0.0}          AuthAADB2CTest.Windows
Microsoft.WindowsAppSDK                     {1.1.0}          AuthAADB2CTest.Windows
```

### Affected platforms

WebAssembly

### IDE

Visual Studio 2022

### IDE version

Microsoft Visual Studio Community 2022 (64-bit) - Preview Version 17.3.0 Preview 3.0

**Comment by jader1313** [comment]
We are having the exact same problem.

**Comment by Hardshop18** [comment]
I'm facing the same reported problem.

**Comment by Hardshop18** [comment]
I did some more tests and the browser that is not able to close the callback popup is Google Chrome. I tested it in Microsoft Edge and Opera and are closing the callback popup.

**Comment by jader1313** [comment]
In my tests problem is debug within Goggle Chrome
